### ISSUE TITLE
Two mentions of the same un-imported identifier cause two prompts to import

### ISSUE THREAD

**Issue by greenrd** [haskell/haskell-mode#564]
There should only be one prompt, after which (if accepted) the problem will be solved.

**Comment by gracjan** [comment]
True. Realistically, this will be low priority for anybody execpt you, @greenrd. Do you have time hack a solution? If not you then it might take ages before somebody else gets to this one.

**Comment by greenrd** [comment]
Yes, I took a quick look at it, but I am not experienced in elisp programming and I am not sure what is the best way to temporarily remember which identifiers were imported, only for the current compilation (I guess they should be forgotten after that, because it might be compiling a different file or a different version of the file).

**Comment by gracjan** [comment]
As usual in such case it is best to just try something out and get the hands dirty.

Interactive mode has a place where it resets things when it knows new compilation has started. Meanwhile you could keep a list of already-done-fixes. Alternativelly you can check is-this-fix-still-needed and offer prompt only when it is still the case.

**Comment by gracjan** [comment]
@greenrd: `haskell-process.el` has `haskell-process-set-suggested-imports` that says

> "Remember what imports have been suggested, to avoid
> re-asking about the same imports."

Doesn't it work for you?

**Comment by gracjan** [comment]
This looks like a bug, because apparently the functionality is there (just not working).

The task is then:
- [ ] create a Haskell source file that triggers two import suggestions for the same module
- [ ] make sure that if the first suggestion was accepted it isn't asked second time

**Comment by gracjan** [comment]
Related #1170.
